### PR TITLE
Multiple action formats

### PR DESCRIPTION
**Pull request by nhunzaker**
Actions once again support promises and returned values. We strictly enforce parameters passed to arguments to be one function, however this should help with the upgrade. I'm comfortable making this a minor release given it does not break actions period, it only makes their usage more strict.

This means you can design actions to follow:

**Return a value:**

```javascript
action: function(params) {
    return params
}
```

**Return a promise:**

```javascript
action: function(params) {
    return new Promise(...)
}
```

**Provide an error-first callback:**

```javascript
action: function(params, next) {
  if (!params.required) {
    next(new Error('Please provide the required field'), null)
  } else {
    next(null, 'stuff')
  }
}
```

### CODE REVIEW

File: src/Microcosm.js [hunk 1]
```diff
@@ -73,20 +74,17 @@ class Microcosm extends Foliage {
     return this
   }
 
-  prepare(action, ...buffer) {
+  prepare(/* action, ...buffer */) {
```
Comment (solomonhawk) [inline src/Microcosm.js]: Forgive the ignorance, what change/benefit is there to doing it this way in comparison to rest params as you had before?

File: src/Signal.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    callbacks.forEach(fn => fn(error, result))
+  })
+}
+
+function passThrough (action, params, next, callbacks) {
+  let result = action(params)
+
+  // Actions some times return thenables. When this happens, wait for
```
Comment (solomonhawk) [inline src/Signal.js]: `thenable` :+1:

**Comment by solomonhawk** [comment]
:ship:

File: src/Microcosm.js (hunk 1 above)
Comment (nhunzaker) [inline src/Microcosm.js]: It's silly, but the compiled code is smaller. I'll run the numbers, the confusion is not worth 20 bytes

Comment (nhunzaker) [inline src/Microcosm.js]: Uhhh. oddly. The difference is 17 bytes. I'll just keep the spread.

File: src/Signal.js (hunk 2 above)
Comment (nhunzaker) [inline src/Signal.js]: Yeah! We support the Promise API without expecting promises to exist

**Pull request merged**